packages/signals/src/signal.rs

```diff
@@ -237,7 +208,7 @@ impl<T: 'static, S: Storage<SignalData<T>>> Writable for Signal<T, S> {
     }
 
     #[track_caller]
-    fn try_write(&self) -> Result<Self::Mut<T>, generational_box::BorrowMutError> {
+    fn try_write(&mut self) -> Result<Self::Mut<T>, generational_box::BorrowMutError> {
```

Comment:
I was using this method some times as a way to get a writable value without the need to making mut the value, is there a way to avoid adding &mut? Or maybe is it possible to have another method that takes &self?

Reply:
The idea with signals is by default borrow checking within the same scope works like normal. This should give you a compile time error just like normal state:
```rust
let mut signal = Signal::new(0);
let data = signal.read();
signal.set(1);
println!("{data}");
```
But once you copy it into a different scope, it resets any borrows:
```rust
let mut signal = Signal::new(0);
let data_callback = move || signal.read();
let data = data_callback();
signal.set(1);
println!("{data}");
```

Two issues currently exist:
1) We don't thread the lifetimes through the generational borrows, so both of the above work
2) The easy way to opt out of this checking was removed during some other updates. We used to support `write_unchecked` which took `&self` instead of `&self`. Right now, you can make a copy of the signal and write to that, but it isn't as easy:
let signal = Signal::new(0);
let signal_borrow = &signal;
let mut copied = signal;
// You can write to the copy while you hold a reference to signal
copied.set(1);
println!("{signal_borrow}");
```

Reply:
Regarding the .2, couldn't there be something like `with_mut` that actually takes `&self` but gives you `&mut T` by giving you a mutable reference to a copy of the signal? or something like that, maybe calling it `with_unchecked`

Reply:
Yes, either `with_unchecked` or `write_unchecked` would work

Reply:
> Yes, either `with_unchecked` or `write_unchecked` would work

Maybe I can implement this myself

Reply:
`with_unchecked`? Yes with https://github.com/DioxusLabs/dioxus/pull/2022 you can implement that pretty easily

Reply:
@marc2332 in the short term you can re-ref which will give you a copy and relax the borrow checker.